### ISSUE TITLE
`AssertArrayContainsOnly` doesn't seem to work...

### ISSUE THREAD

**Issue by bvisin** [rokucommunity/rooibos#10]
![image](https://user-images.githubusercontent.com/26719086/45983970-ce214a80-c013-11e8-853b-ceefce2faf83.png)
```
Brightscript Debugger> ?methodName
RBS_CMN_IsArray

Brightscript Debugger> ?type(methodName)
String

Brightscript Debugger> ?methodName(item)
Function Call Operator ( ) attempted on non-function. (runtime error &he0) in $LIVECOMPILE(48)

Brightscript Debugger> ?RBS_CMN_IsArray(item)
true
```

**Comment by bvisin** [comment]
I'm not sure how to get brightscript call functions by string names...

**Comment by georgejecook** [comment]
this code here is a throwback to the original unit testing library. I hate this function. It's badly named imo, which I inherited. It should've been AssertArrayContainsItemsOfType.
The code is indeed broken though - the methodNAme is right; but it's missing a call to `RBS_CMN_GetFunction`

I'll fix this tonight/tomororw am (ET)

**Comment by georgejecook** [comment]
here's the fix I need to test it

```Function RBS_BTS_AssertArrayContainsOnlyObjectsOfType(array as dynamic, typeStr as string, msg = "" as string) as dynamic
    if (m.currentResult.isFail) then return m.GetLegacyCompatibleReturnValue(false) ' skip test we already failed
    if RBS_CMN_IsAssociativeArray(array) or RBS_CMN_IsArray(array)
        methodName = "RBS_CMN_Is" + typeStr
        typeCheckFunction = RBS_CMN_GetFunction(invalid methodName)
        if (typeCheckFunction <> invalid)
            for each item in array
                if not typeCheckFunction(item)
                    msg = RBS_CMN_AsString(item) + "is not a '" + typeStr + "' type."
                    m.currentResult.AddResult(msg)
                    return m.GetLegacyCompatibleReturnValue(false)
                end if    
            end for
        else
            msg = "could not find comparator for type '" + typeStr + "' type."
            m.currentResult.AddResult(msg)
            return m.GetLegacyCompatibleReturnValue(false)    
        end if
    else
        msg = "Input value is not an Array."
        m.currentResult.AddResult(msg)
        return m.GetLegacyCompatibleReturnValue(false)
    end if
    m.currentResult.AddResult("")
    return m.GetLegacyCompatibleReturnValue(true)
End Function```

**Comment by georgejecook** [comment]
Resolved in commit 06fa24c
